src/buttons/EditNavMap.tsx

```diff
@@ -0,0 +1,122 @@
+import { Box, Button, Stack } from "@mui/material";
+import { useState } from "react";
+
+import { Photosphere, VFE } from "../DataStructures";
+import PhotosphereSelector, {
+  PhotosphereSelectorProps,
+} from "../PhotosphereSelector";
+
+interface EditNavMapProps {
+  onClose: () => void;
+  vfe: VFE;
+  onUpdateVFE: (updatedPhotospheres: Record<string, Photosphere>) => void;
+}
+
+function EditNavMap({ onClose, vfe, onUpdateVFE }: EditNavMapProps) {
+  const [selectedPhotosphere, setSelectedPhotosphere] = useState("");
+  const [localPhotospheres, setLocalPhotospheres] = useState<
+    Record<string, Photosphere>
+  >(vfe.photospheres);
+
+  const options: string[] = Object.keys(vfe.photospheres);
+
+  const selectorProps: PhotosphereSelectorProps = {
+    options: options,
+    value: selectedPhotosphere,
+    setValue: setSelectedPhotosphere,
+  };
+
+  const map = vfe.map;
+
+  function handleMapClick(event: React.MouseEvent<HTMLDivElement>) {
+    if (!selectedPhotosphere || !map) {
+      alert("Please select a photosphere and ensure a map is loaded first.");
+      return;
+    }
+
+    const rect = (event.target as HTMLDivElement).getBoundingClientRect();
+    const x = event.clientX - rect.left;
+    const y = event.clientY - rect.top;
+
+    const updatedPhotospheres = { ...localPhotospheres };
+    const photosphere = updatedPhotospheres[selectedPhotosphere];
+    photosphere.center = { x: x * 2, y: y * 2 };
```

Comment:
What's the reason for multiplying by 2? A comment explaining it would be helpful.